### PR TITLE
Offer to install Java in case of issues

### PR DESCRIPTION
**Pull request by tgodzik**
Also, made a fix to only update the workspace config if that one was in fact invalid. This might be helpful in case of remote workspaces, where globally updating the config will break the local settings.

It would be ebst in those cases to update the remote setting, but it doesn't seem to be possible currently.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -266,17 +285,25 @@ function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
...
-        if (choice === openSettingsAction) {
-          commands.executeCommand(openSettingsCommand);
-        }
-      });
+      window
+        .showErrorMessage(msg, openSettingsAction, downloadJava)
+        .then((choice) => {
+          if (choice === openSettingsAction) {
+            commands.executeCommand(openSettingsCommand);
+          } else if (choice == downloadJava) {
```
Comment (gabro) [inline src/extension.ts:293]:
```suggestion
          } else if (choice === downloadJava) {
```

**Review by gabro — APPROVED** [review]
One super minor thing, but otherwise LGTM

File: src/extension.ts (hunk 1 above)
Comment (tgodzik) [reply src/extension.ts]: Didn't notice that one, thanks!

**Review by dos65 — COMMENTED** [review]
I've seen that there is a usage o `locate-java-home` library inside the `getJavaConfig` and we use it to check the version of JDK. I'm wondering why it doesn't fail if there is no JDK installation.
https://github.com/scalameta/metals-vscode/blob/3c4e460ef20ff6504e1d8a4e3cbcae8e20a0abcc/src/extension.ts#L231-L236

**Comment by tgodzik** [comment]
> I've seen that there is a usage o `locate-java-home` library inside the `getJavaConfig` and we use it to check the version of JDK. I'm wondering why it doesn't fail if there is no JDK installation.
> https://github.com/scalameta/metals-vscode/blob/3c4e460ef20ff6504e1d8a4e3cbcae8e20a0abcc/src/extension.ts#L231-L236

It does fail, but not if we manually configured java-home or java has a wrong version. I this case it might fail while trying to start coursier or Metals. In that case it's nice to give the users a possibility to download it automatically I think.

**Review by dos65 — APPROVED** [review]
> It does fail, but not if we manually configured java-home or java has a wrong version. I this case it might fail while trying to start coursier or Metals. In that case it's nice to give the users a possibility to download it automatically I think.

Yeah, I think it's good fallback option.

Btw, I found the original issue where user said that he has JDK 16 but it looks like extension shouldn't accept it (according [to this line](https://github.com/scalameta/metals-languageclient/blob/main/src/getJavaHome.ts#L51)). We either use this library improperly or there is a bug.

Also, I wasn't clear that it's just question.
LGTM!

**Pull request merged**